pokemon-showdown

```diff
@@ -16,15 +18,18 @@ try {
 try {
 	require.resolve('sockjs');
 } catch (e) {
-	let command = 'npm install --production';
-	console.log('Installing dependencies: `' + command + '`...');
-	require('child_process').spawnSync('sh', ['-c', command], {stdio: 'inherit'});
+	console.log('Installing dependencies...');
+	spawnSync('npm', ['install', '--production'], {stdio: 'inherit', shell: true});
 }
 
-// start the server
+// Start the server
 
 let port;
 if (process.argv[2]) {
 	port = parseInt(process.argv[2]); // eslint-disable-line radix
 }
-require('./app').listen(port);
+
+// Bootstrap manually to ensure any dependencies that were just installed can
+// be found when running on Windows
+
+require('module')._load('./app', module, false).listen(port);
```

Comment:
The ``module`` system is pretty much frozen even regarding undocumented features,
since any change in Node core would be prevented due to the widespread usage of undocumented APIs in the ecosystem.

So yes, this is probably stable. However, this use case looks very suspect to me...

At any rate, shouldn't this call be ``require('module')._load('./app', module, true).listen(port);``,
so that ``app.js`` is effectively treated as the main module as it is expected?